### DISCUSSION TITLE
Naming stack local variables instead of using the hard to recognize offsets.

### DISCUSSION THREAD

**Discussion by itewqq** [Vector35/binaryninja-api#5102] [Working with Decompilation]
I'm new to binja, but I'm finding that both the pseudo-C code and the High level IR have a large number of stack pointer offset based results that are very difficult to sort out in the code:
![image](https://github.com/Vector35/binaryninja-api/assets/30570177/b4dd0978-d7f1-4583-8230-115459cbc0bc)

Is there a good way to rename them or make them local variables? In IDA we can change the function property to BP-Based Frame and it will automatically gives a label to each stack variable. How to make it happen in Binja?
![image](https://github.com/Vector35/binaryninja-api/assets/30570177/0651d696-aa55-4103-9a2c-c1303ec66494)

**Comment by plafosse** [comment]
You should be able to do this but I think in this case however we've failed to resolve the stack pointer correctly. Its hard to tell what the issue is here without seeing more of the function. Could you possibly share the binary?

**Comment by itewqq** [comment]
@plafosse Sure, just a regular ctf-style RE chall: https://1drv.ms/u/s!AlfI-PgV_8WnjdwdpPn6Cx8xiQMSlg?e=ArOUl0

**Comment by plafosse** [comment]
Ok yeah the problem here is that the stack alignment primitive screws up our stack analysis. We should be resilient to this. You can work around this problem by nopping out the stack alignment

If you go to LowLevelIL and turn on "Show Stack Pointer Value"
<img width="492" alt="image" src="https://github.com/Vector35/binaryninja-api/assets/10556585/c64e6da9-a132-4968-82c5-ec71c5313d75">

You will see that we loose track of rsp on the stack alignment instruction.

<img width="655" alt="image" src="https://github.com/Vector35/binaryninja-api/assets/10556585/e5453952-8a97-49dc-9826-5291b1765b14">

If you nop out that instruction it looks good:
<img width="427" alt="image" src="https://github.com/Vector35/binaryninja-api/assets/10556585/66cc88fa-24fb-47e0-868a-0465ae78abfe">

**Comment by plafosse** [comment]
I'm actually going to convert this to an issue.
